### ISSUE TITLE
Assertion `pjsip_rdata_get_tsx(rdata)==((void*)0)' failed occured when using sipecho 

### ISSUE THREAD

**Issue by Kexin6** [pjsip/pjproject#3368]
### Describe the bug

This bug occurs when the SIP client sends a malicious INVITE message with a malformed IPv6 address, and sipecho attempts to respond. 

### Steps to reproduce

1. Compile PJSIP binaries under pjproject directory: `./configure --disable-strip CFLAGS=-g && make dep && make clean && make`

2. Compile Baresip binary under baresip directory: `make -C deps/re install PREFIX=$PWD/bin && make -C deps/rem install PREFIX=$PWD/bin LIBRE_MK=$SIP_FUZZER_DIR/baresip/deps/re/mk/re.mk && make PREFIX=$PWD/bin LIBRE_MK=deps/re/mk/re.mk LIBRE_INC=bin/include/re/ LIBRE_SO=bin/lib/ LIBREM_PATH=bin/ install`

3. Launch SIP Echo service: `./pjproject/pjsip-apps/bin/samples/x86_64-unknown-linux-gnu/sipecho`

4. Launch Baresip client:` ./baresip -f docs/examples -e \"d test@127.0.0.1:5060\" -n 127.0.0.1"`

5. Baresip sends the following INVITE message to the SIP Echo application with an invalid IPv6 origin address:

```
	INVITE sip:test@127.0.0.1:5060 SIP/2.0\r\n
	Via: SIP/2.0/UDP 127.0.0.1:40670;branch=z9hG4bKc15522e59a81c423;rport\r\n
	Contact: <sip:user-0x276d010@127.0.0.1:40670>\r\n
	Max-Forwards: 70\r\n
	To: <sip:test@127.0.0.1:5060>\r\n
	From: <sip:user@example.com>;tag=b6976b1b164c9bb8\r\n
	Call-ID: 1dec19e183a3371f\r\n
	CSeq: 3677 INVITE\r\n
	User-Agent: baresip v0.6.2 (x86_64/linux)\r\n
	Allow: INVITE,ACK,BYE,CANCEL,OPTIONS,NOTIFY,SUBSCRIBE,INFO,MESSAGE,REFER\r\n
	Supported: gruu\r\n
	Content-Type: application/sdp\r\n
	Content-Length: 308\r\n\r\n
	v=0\r\n
	o=- 200339302 1210858333 IN IP6 ?\r\n
	s=-\r\n
	c=IN IP6 ?\r\n
	t=0 0\r\n
	a=tool:baresip 0.6.2\r\n
	m=audio 12870 RTP/AVP 0 8 101\r\n
	a=rtpmap:0 PCMU/8000\r\n
	a=rtpmap:8 PCMA/8000\r\n
	a=rtpmap:101 telephone-event/8000\r\n
	a=fmtp:101 0-15\r\n
	a=sendrecv\r\n
	a=label:1\r\n
	a=rtcp-rsize\r\n
	a=ssrc:1144089027 cname:sip:user@example.com\r\n
	a=ptime:20\r\n
```

6. SIP Echo application hits the assertion error `'pjsip_rdata_get_tsx(rdata)==((void*)0)' failed` in `sip_util.c`

### PJSIP version

2.13, 2.8

### Context

This assertion indicates that the incoming message has a corresponding transaction instance. The error occurs in the `sipecho.c` file, specifically in the following function call on line 532:
```
if (status != PJ_SUCCESS) {
  pjsip_endpt_respond_stateless( app.sip_endpt, rdata,
   500, NULL, NULL, NULL);
}
```

Instead of causing an assertion, sipecho should first check if the incoming message has a UAS transaction before deciding to call `pjsip_endpt_respond_stateless()` or `pjsip_endpt_respond()`. The `pjsip_endpt_respond_stateless()` function is typically used for sending non-transactional responses such as 1xx, 2xx, 3xx and 4xx. While a 500 message can be stateless, inserting a  check beforehand can avoid the assertion and gracefully handle the error.

I have attached a potential fix here. Please review it to see if it makes sense. Thanks!

[0001-sipecho-assertion-fix-if-incoming-message-has-transaction-instance.patch](https://github.com/pjsip/pjproject/files/10671303/0001-sipecho-assertion-fix-if-incoming-message-has-transaction-instance.patch)

### Log, call stack, etc

```shell
#1  0x000000000043fb1b in pjsip_endpt_respond_stateless (endpt=0x16b4378, rdata=0x16c95a8, st_code=500, st_text=0x0, hdr_list=0x0, body=<optimized out>)
    at ../src/pjsip/sip_util.c:1861
#2  0x0000000000405e05 in on_rx_request (rdata=0x16c95a8) at ../src/samples/sipecho.c:532
#3  0x000000000043a337 in pjsip_endpt_process_rx_data (endpt=<optimized out>, rdata=0x16c95a8, p=0x7f71bb9a5410, p_handled=<optimized out>)
    at ../src/pjsip/sip_endpoint.c:893
#4  0x00000000004390df in endpt_on_rx_msg (endpt=0x16b4378, status=0, rdata=0x16c95a8) at ../src/pjsip/sip_endpoint.c:1043
#5  0x0000000000449563 in pjsip_tpmgr_receive_packet (mgr=<optimized out>, rdata=0x16c95a8) at ../src/pjsip/sip_transport.c:1978
#6  0x000000000044d623 in udp_on_read_complete (key=<optimized out>, op_key=<optimized out>, bytes_read=<optimized out>)
    at ../src/pjsip/sip_transport_udp.c:192
#7  0x0000000000491548 in ioqueue_dispatch_read_event (ioqueue=0x16bf150, h=0x16bfc40) at ../src/pj/ioqueue_common_abs.c:605
#8  0x0000000000495569 in pj_ioqueue_poll (ioqueue=<optimized out>, timeout=<optimized out>) at ../src/pj/ioqueue_select.c:1070
#9  0x0000000000439bbc in pjsip_endpt_handle_events2 (endpt=0x16b4378, max_timeout=0x7f71bb9a5f10, p_count=0x0) at ../src/pjsip/sip_endpoint.c:744
#10 0x0000000000404925 in worker_proc (arg=<optimized out>) at ../src/samples/sipecho.c:167
#11 0x0000000000497390 in thread_main (param=0x16b4168) at ../src/pj/os_core_unix.c:541
#12 0x00007f71bc18e6ba in start_thread (arg=0x7f71bb9a6700) at pthread_create.c:333
#13 0x00007f71bbec451d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109

sipecho: ../src/pjsip/sip_util.c:1861: 
pj_status_t pjsip_endpt_respond_stateless(pjsip_endpoint *, pjsip_rx_data *, 
int, const pj_str_t *, const pjsip_hdr *, const pjsip_msg_body *): 
Assertion `pjsip_rdata_get_tsx(rdata)==((void*)0)' failed.
```

**Comment by trengginas** [comment]
I can't reproduce it here. The pjsip_get_sdp_info() will return error and will not get passsed https://github.com/pjsip/pjproject/blob/master/pjsip-apps/src/samples/sipecho.c#L467.
Please make sure you are using the latest code.

**Comment by sauwming** [comment]
Also, we'd recommend you use pjsua sample app to reproduce any issue, instead of sipecho. The sample apps in the `samples` directory are considered small sample apps, usually for proof-of-concept purpose of a certain feature, such as AEC, jitter buffer, etc. These small apps are not meant to be fully robust and do not have the most comprehensive error checking.

**Comment by Kexin6** [comment]
Thanks for the suggestion!  As for this case, I was testing on the latest version in a clean docker and here is the log:
```
17:54:14.007         os_core_unix.c !pjlib 2.13-dev for POSIX initialized
17:54:14.008            udp0xca6910  SIP UDP transport started, published address is 172.17.0.2:5060
Ready for incoming calls. Press ENTER to quit...

Menu:
  h    Hangup all calls
  l    Enable message logging
  q    Quit

17:54:14.154              sipecho.c !.RX INVITE from 127.0.0.1
17:54:14.154              sipecho.c  .Call 0: creating answer:
17:54:14.154              sipecho.c  .  Media 0, audio: sendrecv <--> ?:16836
PRELOAD SERVER inside sendto, len: 3896
17:54:14.154      tsx0x7fa540002d88  ....Error sending Response msg 180/INVITE/cseq=45495 (tdta0x7fa54000c1c8): Message too long (PJSIP_EMSGTOOLONG)
17:54:14.155      tsx0x7fa540002d88  ....Failed to send Response msg 180/INVITE/cseq=45495 (tdta0x7fa54000c1c8)! err=171023 (Message too long (PJSIP_EMSGTOOLONG))

sipecho: ../src/pjsip/sip_util.c:1881: pj_status_t pjsip_endpt_respond_stateless(pjsip_endpoint *, pjsip_rx_data *, int, const pj_str_t *, const pjsip_hdr *, const pjsip_msg_body *): Assertion `pjsip_rdata_get_tsx(rdata)==((void*)0)' failed.
```
I suspect that this assertion error was triggered by server trying to send a message with a maliciously long Branch ID `branch=z9hG4bKa33f44d244edad9%3a%3a%3a%3a%3a%3a%3a%3a%3a%3a%3a%3a%3a%3a...%3a <repeated>`.

I managed to capture the packets to the [assert.pcap](https://github.com/Kexin6/pjproject/blob/master/pcap_files/assert.pcap) file and the human-readable [beautiful_assert.txt](https://github.com/pjsip/pjproject/files/10702189/beautiful_assert.txt). You may see from the .pcap file that the server attempted to send the very long 100 Trying message as a reply to the very long client message.

I hope I clarified the triggering situation this time. Thank you for taking the close look.
